src/driver/KoaDriver.ts

```diff
@@ -161,8 +161,20 @@ export class KoaDriver extends BaseDriver implements Driver {
             case ParamTypes.PARAM:
                 return context.params[param.name];
             case ParamTypes.SESSION:
-                throw new Error("@Session decorator is not supported by KoaDriver yet.");
-                // TODO return session
+                if (context.session === undefined) {
+                    throw new Error("Session is undefined! Is any Session-Middleware exist?");
```

Comment:
Throwing Error here is a bad idea, it should be removed and just returned undefined or not. Checking if the param is empty is already [here](https://github.com/pleerock/routing-controllers/blob/master/src/ParamHandler.ts#L45).